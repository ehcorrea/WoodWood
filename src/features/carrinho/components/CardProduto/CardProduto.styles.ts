import styled, { css } from '@emotion/native';

export const Wrapper = styled.View`
  ${({ theme }) => css`
    align-items: center;
    background-color: ${theme.colors.primary[20]};
    border-radius: 20px;
    flex-direction: row;
    height: 130px;
    padding: 20px;
  `}
`;

export const Image = styled.Image`
  aspect-ratio: 1;
  height: 100%;
`;

export const ContainerInfos = styled.View`
  flex: 1;
  height: 100%;
  justify-content: center;
`;

export const ContainerButtons = styled.View`
  border-color: black;
  border-radius: 20px;
  border-width: 2px;
`;

export const Line = styled.View`
  border-color: black;
  border-width: 1px;
`;
