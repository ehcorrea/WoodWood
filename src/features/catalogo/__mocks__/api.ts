import { mocked } from 'jest-mock';
import * as api from '../api/api';

jest.mock('../api/api');

export const mockedApi = mocked(api);
