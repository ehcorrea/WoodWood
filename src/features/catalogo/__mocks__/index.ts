export * from './api';
export * from './produtos';
