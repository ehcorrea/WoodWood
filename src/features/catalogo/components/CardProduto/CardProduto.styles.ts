import styled, { css } from '@emotion/native';

import { Text } from '@/shared/components';

export const Wrapper = styled.View`
  ${({ theme }) => css`
    align-items: center;
    background-color: ${theme.colors.white.main};
    border-radius: 10px;
    elevation: 3;
    flex: 1;
    min-height: 230px;
    justify-content: space-between;
    margin-horizontal: 5px;
    shadow-color: #000;
    shadow-offset: 1px 3px;
    shadow-opacity: 0.2;
    shadow-radius: 2px;
    overflow: hidden;
  `}
`;

export const ContainerPreco = styled.View`
  position: absolute;
  top: 10%;
  left: -28%;
  transform: rotate(-47deg);
  z-index: 1;
  background-color: yellow;
  width: 100%;
  min-height: 10%;
  padding: 5px 0;
`;

export const Preco = styled(Text)`
  text-align: center;
`;

export const Image = styled.Image`
  aspect-ratio: 1;
  margin: 10px;
  height: 50%;
`;

export const Titulo = styled(Text)`
  padding: 0 10px;
`;

export const FakeButton = styled.View`
  ${({ theme }) => css`
    align-items: center;
    background-color: ${theme.colors.primary.main};
    justify-content: center;
    min-height: 30px;
    padding: 2px 5px;
    width: 100%;
  `}
`;
