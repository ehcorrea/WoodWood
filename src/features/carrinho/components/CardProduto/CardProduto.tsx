import { TouchableOpacity } from 'react-native';

import { ProdutoDoCarrinho } from '@/shared/types';
import { Spacing, Text } from '@/shared/components';

import { IconAdd, IconRemove } from '../../assets/icons';

import * as S from './CardProduto.styles';

type CardProdutoProps = {
  produto: ProdutoDoCarrinho;
  onAdicionar: (prduto: ProdutoDoCarrinho) => void;
  onRemover: (prduto: ProdutoDoCarrinho) => void;
};

export function CardProduto({
  produto,
  onAdicionar,
  onRemover,
}: CardProdutoProps) {
  const handlePressRemover = () => {
    onRemover(produto);
  };

  const handlePressAdicionar = () => {
    onAdicionar(produto);
  };

  return (
    <S.Wrapper>
      <S.Image
        source={{ uri: produto.image }}
        testID="image"
        resizeMode="contain"
      />
      <Spacing x={5} />
      <S.ContainerInfos>
        <Text.Subtitle numberOfLines={1}>{produto.title}</Text.Subtitle>
        <Spacing y={5} />
        <Text numberOfLines={2}>
          {produto.quantidade} x R$ {produto.price.toFixed(2)}
        </Text>
        <Spacing y={5} />
        <Text numberOfLines={2}>
          Total:{' '}
          <Text.Subtitle numberOfLines={1}>
            R${produto.total.toFixed(2)}
          </Text.Subtitle>
        </Text>
      </S.ContainerInfos>
      <Spacing x={5} />
      <S.ContainerButtons>
        <TouchableOpacity
          accessibilityLabel={`Adicionar mais uma unidade de ${produto.title}`}
          onPress={handlePressAdicionar}
        >
          <IconAdd width={35} height={35} />
        </TouchableOpacity>
        <S.Line />
        <TouchableOpacity
          accessibilityLabel={`Remover uma unidade de ${produto.title}`}
          onPress={handlePressRemover}
        >
          <IconRemove width={35} height={35} />
        </TouchableOpacity>
      </S.ContainerButtons>
    </S.Wrapper>
  );
}
