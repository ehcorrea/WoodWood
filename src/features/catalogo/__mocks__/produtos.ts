import { Produto } from '../api/types';

export const MOCKED_PRODUTOS: Produto[] = [
  {
    category: 'mocked-category',
    description: 'mocked-description',
    id: 0,
    image: 'mocked-image',
    price: 100,
    rating: {
      count: 10,
      rate: 10,
    },
    title: 'mocked-title',
  },
];
