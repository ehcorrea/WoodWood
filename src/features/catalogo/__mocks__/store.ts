import { mockedStore } from '@/shared/__mocks__';

import { catalogoStore, CatalogoStore } from '../store';

export const mockedCatalagoStore = (storeValues?: Partial<CatalogoStore>) =>
  mockedStore({
    store: catalogoStore,
    storeValues: { ...catalogoStore.getState(), ...storeValues },
  });
