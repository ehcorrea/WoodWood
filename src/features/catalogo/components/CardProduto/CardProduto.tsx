import { Produto } from '@/services/api';
import { Text } from '@/shared/components';

import * as S from './CardProduto.styles';

type CardProdutoProps = {
  produto: Produto;
};

export function CardProduto({ produto }: CardProdutoProps) {
  return (
    <S.Wrapper>
      <S.ContainerPreco>
        <S.Preco weight="semibold" maxFontSizeMultiplier={1}>
          R$ {produto.price}
        </S.Preco>
      </S.ContainerPreco>
      <S.Image source={{ uri: produto.image }} resizeMode="contain" />
      <S.Titulo numberOfLines={2}>
        {produto.title}
        {'\n'}
      </S.Titulo>
      <S.FakeButton>
        <Text palette="white" size="large" weight="semibold">
          Ver detalhes
        </Text>
      </S.FakeButton>
    </S.Wrapper>
  );
}
