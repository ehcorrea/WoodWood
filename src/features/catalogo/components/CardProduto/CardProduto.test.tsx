import { render } from '@/test/utils';
import { criarMockProduto } from '@/shared/__mocks__';

import { CardProduto } from './CardProduto';

const produto = criarMockProduto();

describe('<CardProduto/>', () => {
  test('quando renderizado', () => {
    const container = render(<CardProduto produto={produto} />);
    expect(container.getByText(produto.title)).toBeTruthy();
    expect(container.getByText(`R$ ${produto.price}`)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
