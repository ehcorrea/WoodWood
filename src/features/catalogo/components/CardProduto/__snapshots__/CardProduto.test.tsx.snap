// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardProduto/> quando renderizado 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "elevation": 3,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
        "marginHorizontal": 5,
        "minHeight": 230,
        "overflow": "hidden",
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 3,
          "width": 1,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "yellow",
          "left": "-28%",
          "minHeight": "10%",
          "paddingBottom": 5,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 5,
          "position": "absolute",
          "top": "10%",
          "transform": [
            {
              "rotate": "-47deg",
            },
          ],
          "width": "100%",
          "zIndex": 1,
        },
        undefined,
      ]
    }
  >
    <Text
      maxFontSizeMultiplier={1.4}
      style={
        [
          {
            "color": "#1A1D26",
            "fontFamily": "SofiaPro-SemiBold",
            "fontSize": 21.5,
            "lineHeight": 25.5,
          },
          [
            {
              "textAlign": "center",
            },
            undefined,
          ],
        ]
      }
      weight="semibold"
    >
      R$ 
      100
    </Text>
  </View>
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "mocked-image",
      }
    }
    style={
      [
        {
          "aspectRatio": 1,
          "height": "50%",
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
        undefined,
      ]
    }
  />
  <Text
    numberOfLines={2}
    style={
      [
        {
          "color": "#1A1D26",
          "fontFamily": "SofiaPro-Regular",
          "fontSize": 21.5,
          "lineHeight": 25.5,
        },
        [
          {
            "paddingBottom": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 0,
          },
          undefined,
        ],
      ]
    }
  >
    mocked-title
    

  </Text>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#EA5B0C",
          "justifyContent": "center",
          "minHeight": 30,
          "paddingBottom": 2,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 2,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      numberOfLines={1}
      palette="white"
      size="large"
      style={
        [
          {
            "color": "#FFFFFF",
            "fontFamily": "SofiaPro-SemiBold",
            "fontSize": 25.5,
            "lineHeight": 28.5,
          },
          undefined,
        ]
      }
      weight="semibold"
    >
      Ver detalhes
    </Text>
  </View>
</View>
`;
